util/json/unmarshal.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package json provides json utilities to improve performance.
+// It includes an Unmarshal wrapper that amortizes allocated garbage over subsequent runs
+// and a Bytes type to reduce allocations when unmarshalling a non-hex-encoded string into a []byte.
+package json
```

Comment:
What's the plan with this name?

If we plan to stop using encoding/json everywhere else, then this name makes sense (code looks the same)

If we're going to be using both the foreseeable future, then this gets ugly quickly with people renaming them on import all the time. (and perhaps inconsistently)

Unless we plan on dropping encoding/json everywhere soon, I'd go for `jsonutil` or `tsjson` (except we already have `hujson` which is kinda our "TS JSON" already)

Reply:
Yeah, I went back and forth about this. I'll rename to `jsonutil`.